lib/ex_unit/lib/ex_unit/case.ex

```diff
@@ -227,6 +229,42 @@ defmodule ExUnit.Case do
   Keep in mind that all tests are included by default, so unless they are
   excluded first, the `include` option has no effect.
 
+  ## Parameterized tests
+
+  Sometimes you want to run the same tests but with different parameters.
+  In ExUnit, it is possible to do by returning a `:parameterize` key in
+  your `setup_all` context. The value must be a list of maps which will be
+  the parameters merged into the test context.
+
+  For example, Elixir has a module called `Registry`, which can have type
+  `:unique` or `:duplicate`, and can control its concurrency factor using
+  the `:partitions` option. If you have a number of tests that *behave the
+  same* across all of those values, you can parameterize those tests with:
+
+      setup_all do
+        parameters =
+          for kind <- [:unique, :duplicate],
+              partitions <- [1, 8],
+              do: %{kind: kind, partitions: partitions}
+
+        [parameterize: parameters]
+      end
+
+  Then, in your tests, you can access the parameters as part of the context:
+
+      test "starts a registry", %{kind: kind, partitions: partitions} do
+        ...
+      end
+
+  Use parameterized tests with care:
+
+    * Abuse of parameterized tests may make your test suite considerably slower
```

Comment:
This would be ideal for adapter tests ([or engines, like this](https://github.com/sorentwo/oban/blob/main/test/oban/engine_test.exs)). Could parameterized tests be parallelized? If not, that's a potential disadvantage compared to generating separate test cases like the linked example.

It seems like parameterized tests may make your test suite slower for two reasons:

1. Lacking `async` parallelization
2. Encouraging running more tests than you normally would, e.g. with a range of values like in the registry example